contributor-guide/modules/ROOT/pages/design-guide/dependencies.adoc

```diff
@@ -3,74 +3,46 @@
 :idprefix:
 :idseparator: -
 
-*Legacy content, currently being updated.*
+A Boost library _should generally not_ depend on other libraries, except for Boost or the pass:[C++] Standard Library. There are some exceptions. For example, boost:iostreams[] has an optional dependency on https://www.zlib.net/[*zlib*], to provide support for gzip and zlib compression and decompression. If zlib is not available, boost:iostreams[] will still work, but without the ability to handle gzip and zlib formats. Another example is boost:asio[], which has an optional dependency on https://www.openssl.org/[*OpenSSL*], to provide support for SSL and TLS protocols. These protocols are widely used for secure network communications. Again, if OpenSSL is not available, boost:asio[] will still work, but without support for SSL or TLS.
 
-A Boost library **should not** use libraries
- other than Boost or the C++ Standard Library.
+Taking a dependency on https://www.zlib.net/[*zlib*] or https://www.openssl.org/[*OpenSSL*] is fine. boost:python[] interfaces with the Python C API, so obviously depends on a Python installation. boost:locale[] depends on the International Components for Unicode (https://icu.unicode.org/[*ICU*]) to provide Unicode and localization support. On Unix systems, boost:thread[] replies on the https://www.cs.cmu.edu/afs/cs/academic/class/15492-f07/www/pthreads.html[*pthread*] library.
```

Comment:
I think I ended up explaining the motivation above, which is basically no unnecessary dependencies.